x-pack/plugin/esql/src/test/java/org/elasticsearch/xpack/esql/expression/function/RailRoadDiagram.java

```diff
@@ -49,18 +49,28 @@ static String functionSignature(FunctionDefinition definition) throws IOExceptio
         List<Expression> expressions = new ArrayList<>();
         expressions.add(new SpecialSequence(definition.name().toUpperCase(Locale.ROOT)));
         expressions.add(new Syntax("("));
-        boolean first = true;
-        List<String> args = EsqlFunctionRegistry.description(definition).argNames();
-        for (String arg : args) {
-            if (arg.endsWith("...")) {
-                expressions.add(new Repetition(new Sequence(new Syntax(","), new Literal(arg.substring(0, arg.length() - 3))), 0, null));
-            } else {
-                if (first) {
-                    first = false;
+
+        if (definition.name().equals("case")) {
+            // CASE is so weird let's just hack this together manually
+            Sequence seq = new Sequence(new Literal("condition"), new Syntax(","), new Literal("trueValue"));
+            expressions.add(new Repetition(seq, 1, null));
+            expressions.add(new Repetition(new Literal("elseValue"), 0, 1));
```

Comment:
I think CASE is the only function that has this funny signature. It's actually pretty helpful to see the railroad diagram for CASE too!